mazeGenerator/mazeGenerator.go

```diff
@@ -0,0 +1,108 @@
+package mazeGenerator
+
+import (
+	"fmt"
+	"math/rand/v2"
+	"zombieApocalypeSOMAS/physicsEngine"
+)
+
+type Maze [][]int
+
+func (m Maze) Print() {
+	for _, row := range m {
+		for _, elem := range row {
+			fmt.Printf("%3d ", elem) // Adjust the width for alignment (e.g., %3d for 3 spaces)
+		}
+		fmt.Println() // Print a new line at the end of each row
+	}
+}
+
+type MazeGenerator struct {
+	M         int
+	N         int
+	maze      Maze
+	dirs      []physicsEngine.Vector2D
+	generator *rand.Rand
+}
+
+func (mg *MazeGenerator) generateInitialMaze(i, j int) {
+	if mg.isOutOfBounds(i, j) {
+		panic("Exit to maze is outside of dimensions")
+	}
+	array := make([][]int, mg.M)
+	for i := range array {
+		array[i] = make([]int, mg.N)
+		for j := range array[i] {
+			array[i][j] = 1
+		}
+	}
+	array[i][j] = 2
+	mg.maze = array
+}
+
+func CreateMazeGenerator(M, N, exit_i, exit_j int, generator *rand.Rand) *MazeGenerator {
+	if M&N&1 == 0 {
+		panic("Both maze dimensions must be odd")
+	}
+	mazeGen := &MazeGenerator{
+		M:    M,
+		N:    N,
+		maze: nil,
+		dirs: []physicsEngine.Vector2D{
+			{X: 0, Y: 1},
+			{X: 1, Y: 0},
+			{X: 0, Y: -1},
+			{X: -1, Y: 0},
+		},
+		generator: generator,
+	}
+	mazeGen.generateInitialMaze(exit_i, exit_j)
+	return mazeGen
+}
+
+func (mg *MazeGenerator) CreateMaze(entrance_i, entrance_j int) Maze {
+	if mg.isOutOfBounds(entrance_i, entrance_j) {
+		panic("Entrance to maze is outside of dimensions")
+	}
+	state := mg.genMaze(entrance_i, entrance_j)
+	if !state {
+		panic("couldnt generate solvable maze. Did you forget to add exits?")
+	}
+	return mg.maze
+}
```

Comment:
Might be worth passing the entrance and the exit coordinates here. It seems strange passing the exit into the constructor, and then the entrance into this function. Let's set `mg.maze[exit_i][exit_j] = 2` here and simplify `generateInitialMaze()`

Reply:
I agree. Do we want to have an entrance though? Or just spawn humans and zombies in and have 1 exit?